pkg/landscaper/installations/context.go

```diff
@@ -52,29 +67,69 @@ func (o *Operation) DetermineInstallationContext(ctx context.Context) (*Context,
 		if inst.Spec.ComponentDescriptor != nil &&
 			inst.Spec.ComponentDescriptor.Reference != nil &&
 			inst.Spec.ComponentDescriptor.Reference.RepositoryContext == nil {
-			inst.Spec.ComponentDescriptor.Reference.RepositoryContext = o.DefaultRepoContext
+			inst.Spec.ComponentDescriptor.Reference.RepositoryContext = externalCtx.RepositoryContext
 		}
 	}
 
-	// get the parent by owner reference
-	parent, err := CreateInternalInstallation(ctx, o.ComponentsRegistry(), parentInst)
-	if err != nil {
-		return nil, err
-	}
-
 	ctxName := ""
 	if parentInst != nil {
 		ctxName = lsv1alpha1helper.DataObjectSourceFromInstallation(parentInst)
 	}
 
 	return &Context{
 		Name:   ctxName,
-		Parent: parent,
+		Parent: CreateInternalInstallationBase(parentInst),
 		// siblings are all encompassed installation of the parent installation
 		Siblings: CreateInternalInstallationBases(siblingInstallations...),
+		External: externalCtx,
 	}, nil
 }
 
+// ExternalContext defines the internal context with additional enhanced context information.
+type ExternalContext struct {
```

Comment:
I'm a bit lost about what makes a context `internal`or `external` ...
External tho whom?

Reply:
You're right this is confusing.
@Diaphteiros and me already tried to find a suitable naming here.

External in this case means that the context is defined in an external resource.
So in another CR in the cluster.

Do you have a better idea for the naming?

Reply:
you mean to the docstring in line 88?

Reply:
yes, exactly